From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Silvio Traversaro <silvio@traversaro.it>
Date: Mon, 6 Jul 2026 17:40:00 +0200
Subject: [PATCH] Fix build with newer Pinocchio FrameTpl API

Newer Pinocchio exposes the parent joint of a frame as parentJoint
instead of parent.
---
 src/placo/model/robot_wrapper.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/placo/model/robot_wrapper.cpp b/src/placo/model/robot_wrapper.cpp
index 993a5be0..9a2ec2b2 100644
--- a/src/placo/model/robot_wrapper.cpp
+++ b/src/placo/model/robot_wrapper.cpp
@@ -323,7 +323,7 @@ void RobotWrapper::compute_hessians()
 Eigen::MatrixXd RobotWrapper::get_frame_hessian(FrameIndex frame, int joint_v_index)
 {
   // Frame parent joint index
-  pinocchio::JointIndex joint_id = model.frames[frame].parent;
+  pinocchio::JointIndex joint_id = model.frames[frame].parentJoint;
   pinocchio::SE3 T = model.frames[frame].placement;
 
   pinocchio::Tensor<double, 3, 0> H = pinocchio::getJointKinematicHessian(model, *data, joint_id, pinocchio::LOCAL);
